pkg/query-service/app/logs/parser.go

```diff
@@ -247,13 +249,33 @@ func replaceInterestingFields(allFields *model.GetFieldsResponse, queryTokens []
 }
 
 func replaceFieldInToken(queryToken string, selectedFieldsLookup map[string]model.LogField, interestingFieldLookup map[string]model.LogField) (string, error) {
+	op := strings.TrimSpace(operatorRegex.FindString(queryToken))
+	opLower := strings.ToLower(op)
+
```

Comment:
By hack did you mean that `replaceFieldInToken` originally intended to just replace the key (of `key-op-value`) in the token with the correct column name? since `(n)exists` is not a binary operator unlike the others, you had to handle it separately?

Reply:
I think there aren't many options, given the previous implementation. This seems fine. If the parsing had condition preparation based on the operator, it would have been better, but it will be too much work to change now with little to no benefit.

Reply:
Yeah right that was what I meant, yeah since we will be replacing it anyway with new QB there is very low chance any new feature updates will be required here.